source/utils/date.cpp

```diff
@@ -132,7 +117,7 @@ int date::weekday() const
     tm.tm_year = year - 1900;
     std::time_t time = std::mktime(&tm);
 
-    return safe_localtime(time).tm_wday;
+    return detail::localtime_safe(time).tm_wday;
```

Comment:
Can't we just use `tm.tm_wday`? According to the specs, the `tm` structure will be updated on success: see https://en.cppreference.com/w/cpp/chrono/c/mktime.

Reply:
You are right, this should work and would avoid an otherwise unnecessary call to `localtime_safe`. Good point :+1:

Reply:
I added a commit that contains your proposed improvement - thanks! :+1: